### PR TITLE
[RHCLOUD-24854] Init race condition

### PR DESCRIPTION
**Pull request by CodyWMitchell**
## What?
Replace usage of `init()` in the logging, config, and s3 packages.

## Why?
Gives more control over the order packages are initialized and prevents race conditions in the code.

## How?
Change the functions to be explicitly called using `Get()`.

## Testing
Did you add any tests for the change?

## Anything Else?
Any other notes about the PR that would be useful for the reviewer. 

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [X] General Coding Practices

### CODE REVIEW

**Comment by dehort** [comment]
Hey @CodyWMitchell, I think I figured out why this PR is failing (most of the time).  You are setting `cfg.Debug = true` in the tests, which I think is happening so that the [middleware will inject a identity header into the http request](https://github.com/RedHatInsights/export-service-go/blob/main/middleware/user.go#L40).  The main issue that is causing so many failures with this PR is that the middleware is calling `config.Get()` which returns a new instance of a `ExportConfig` object.  Since this is a newly created instance (created by the Get() call), it does not have `Debug` set to true ... so the identity header is not injected into the request.  This causes the Identity header middleware to cause the request to be rejected with a 400.

So the main issue is that the instance of ExportConfig used in the tests and the instance of ExportConfig used in the InjectDebugUserIdentity are not the same.

There are a few more failures happening once that is fixed.  I have not tracked those down yet.  I can help with that next week if needed.

This stuff is tricky.  :)

The `InjectDebugUserIdentity` is a little concerning to me.  I'm afraid someone will set `Debug` to true in stage/prod and then its possible for debug user identities to get injected in.  I think that only happens if there is not a identity header already in the request though...so maybe its not too big of a deal.  I am thinking it might be better to just remove this middleware and modify the tests to explicitly pass in the identity header.

**Review by dehort — DISMISSED** [review]
Looks good to me.  I left a couple of comments.

File: config/config.go [hunk 1]
```diff
@@ -79,10 +76,24 @@ type storageConfig struct {
...
+	if !clowder.IsClowderEnabled() || clowder.LoadedConfig.Database.RdsCa == nil {
+		return
+	}
+
+	if rdsCaPathValue, err := clowder.LoadedConfig.RdsCa(); err != nil {
+		panic(err)
+	} else {
+		rdsCaPath = &rdsCaPathValue
+	}
+}
```
[the remark this answers is not in the document]
Comment (CodyWMitchell) [reply config/config.go]: Thanks, Derek! :+1: Moved it to `Get()` here: [d9e67ac](https://github.com/RedHatInsights/export-service-go/pull/66/commits/d9e67acfe042488c1c942908c65e6ae421a676be)

File: exports/request_resources.go [hunk 2]
```diff
@@ -10,13 +10,15 @@ import (
...
 	"go.uber.org/zap"
 
+	"github.com/redhatinsights/export-service-go/config"
 	ekafka "github.com/redhatinsights/export-service-go/kafka"
 	"github.com/redhatinsights/export-service-go/models"
 )
 
 type RequestApplicationResources func(ctx context.Context, log *zap.SugaredLogger, identity string, payload models.ExportPayload)
 
 func KafkaRequestApplicationResources(kafkaChan chan *kafka.Message) RequestApplicationResources {
```
Comment (dehort) [inline exports/request_resources.go:20]: You could pass the `exportsTopic` in here.  I believe this is called in the `api_server.go` initialization code.

**Review by dehort — APPROVED** [review]
Looks good to me!

**Pull request merged**